dvc/data/stage.py

```diff
@@ -247,6 +247,29 @@ def _get_staging(odb: "ObjectDB") -> "ObjectDB":
     return ReferenceObjectDB(fs, fs_path, state=state)
 
 
+def _load_untyped_dir_hash(
+    odb, fs_path, hash_info, meta, staging, verify: bool = False
+):
+
+    from .tree import Tree
+
+    obj = Tree.load(
+        odb, HashInfo(hash_info.name, hash_info.value.replace(".dir", ""))
+    )
+    obj.hash_info = hash_info
+    meta.nfiles = len(obj)
+
+    staging.add(obj.fs_path, obj.fs, obj.hash_info)
+    for key, (_, entry_hash_info) in obj.iteritems():
+        staging.add(
+            odb.fs.path.join(fs_path, *key),
+            odb.fs,
+            entry_hash_info,
+            verify=verify,
```

Comment:
Doesn't seem like verify is used, it is always False

Reply:
Indeed, originally added this to to some benchmarks, `_load_untyped_dir_hash` is 6.5x faster when `verify=False`, will remove.